### PR TITLE
Rename ui-test-extension to ui-items-providers-test

### PR DESCRIPTION
**Pull request by bsteinbk**
In addition to rename 
 - Updates to only use abstract (non-React) items in GeneralUiItemsProvider
 - Add example of setting up 'slice' of redux state
 -  Added stage that shows toolbuttons that use redux state and provide a widget that shows/hides based on selection.
 - Added stage that show how to set up example content

### CODE REVIEW

**Review by NancyMcCallB — APPROVED** [review]

File: test-apps/ui-items-providers-test/src/ui-items-providers-test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ *   4) `GeneralUiItemsProvider` that provide tool buttons and a status bar item to stages set usage set to "StageUsage.General"
+ *   5) `NetworkTracingFrontstage` that defines a new stage to add to iModelApp.
+ *   6) `NetworkTracingUiItemsProvider` that provide tool buttons and widgets to NetworkTracingFrontstage.
+ *   7) `TestProviderState` that define package specific state properties to add to the apps Redux store.
+ *   8) `CustomFrontstage` that define a frontstage the show an imodel view and a custom content view populated via `SampleCustomContent`
+ *
+ */
+export class UiItemsProvidersTest {
+  private static _initialized = false;
+  /** We'll register the uiItemsProvidersTest.json as the Extension's namespace/ */
```
Comment (calebmshafer) [inline test-apps/ui-items-providers-test/src/ui-items-providers-test.ts:33]: No longer an `Extension's namespace`.

File: test-apps/ui-items-providers-test/src/ui-items-providers-test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+/** UiItemsProvidersTest is a package that adds some user interface to the iModelApp when its initialize method is called.
+ * Included in the sample are:
+ *   1) `SampleTool` showing how implement a tool with a variety to tool settings items.
+ *   2) `GenericLocateTool` that shows how implement a tool that requires user to first locate an element.
+ *   3) `OpenAbstractDialogTool` that shows how to generate a dialog without creating "react" components.
+ *   4) `GeneralUiItemsProvider` that provide tool buttons and a status bar item to stages set usage set to "StageUsage.General"
+ *   5) `NetworkTracingFrontstage` that defines a new stage to add to iModelApp.
+ *   6) `NetworkTracingUiItemsProvider` that provide tool buttons and widgets to NetworkTracingFrontstage.
+ *   7) `TestProviderState` that define package specific state properties to add to the apps Redux store.
+ *   8) `CustomFrontstage` that define a frontstage the show an imodel view and a custom content view populated via `SampleCustomContent`
```
Comment (calebmshafer) [inline test-apps/ui-items-providers-test/src/ui-items-providers-test.ts:28]: nit: /imodel/iModel/

File: test-apps/ui-items-providers-test/src/ui-items-providers-test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    // register to add items to "General" usage stages"
+    UiItemsManager.register(new GeneralUiItemsProvider());
+  }
+
+  public static async initialize(): Promise<void> {
+    if (this._initialized)
+      return;
+
+    /** Register the localized strings for this extension
+     * We'll pass the localization member to the rest of the classes in the Extension to allow them to translate strings in the UI they implement.
```
Comment (calebmshafer) [inline test-apps/ui-items-providers-test/src/ui-items-providers-test.ts:63]: another couple references to Extension here

File: test-apps/ui-items-providers-test/src/ui-items-providers-test.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import { IModelApp } from "@itwin/core-frontend";
+import { providerSlice, TestProviderSliceName } from "./store";
+import { GeneralUiItemsProvider } from "./ui/providers/GeneralUiItemsProvider";
+import { OpenAbstractDialogTool } from "./tools/OpenAbstractModalDialogTool";
+import { NetworkTracingFrontstage } from "./ui/frontstages/NetworkTracing";
+import { CustomFrontstage } from "./ui/frontstages/CustomContent";
+
+/** UiItemsProvidersTest is a package that adds some user interface to the iModelApp when its initialize method is called.
+ * Included in the sample are:
+ *   1) `SampleTool` showing how implement a tool with a variety to tool settings items.
```
Comment (calebmshafer) [inline test-apps/ui-items-providers-test/src/ui-items-providers-test.ts:21]: Maybe throw this list up in the README? That may help make it more discoverable what's in the package.

File: test-apps/ui-items-providers-test/src/ui-items-providers-test.ts (hunk 1 above)
Comment (bsteinbk) [reply test-apps/ui-items-providers-test/src/ui-items-providers-test.ts]: removed all references to  `extension`

File: test-apps/ui-items-providers-test/src/ui-items-providers-test.ts (hunk 2 above)
Comment (bsteinbk) [reply test-apps/ui-items-providers-test/src/ui-items-providers-test.ts]: Updated.

File: test-apps/ui-items-providers-test/src/ui-items-providers-test.ts (hunk 3 above)
Comment (bsteinbk) [reply test-apps/ui-items-providers-test/src/ui-items-providers-test.ts]: updated

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**